tracer/tools/PipelineMonitor/BuildProcessor.cs

```diff
@@ -19,36 +19,42 @@ public BuildProcessor(Build build, TimelineData timeline)
 
     public void Process()
     {
-        if (_build.Status != "completed")
-            return;
-
-        if (!_build.StartTime.HasValue || !_build.FinishTime.HasValue)
+        if (!_build.StartTime.HasValue)
+        {
+            // should never happen but doesn't hurt to leave it.
+            Console.WriteLine("only started trace will be taken into account");
             return;
+        }
 
-        var startTime = _build.StartTime;
-        var endTime = _build.FinishTime;
+        var startTime = _build.StartTime.Value;
+        DateTime endTime;
 
-        if (!startTime.HasValue || !endTime.HasValue)
+        if (!_build.FinishTime.HasValue)
```

Comment:
Isn't this only meant to run once the build has finished 🤔 Seems like a bug in our CI setup if this is happening (though not a bug in this code per-se)

Reply:
I wanted to run it once the build is fnished. But as I didn't find a way to trigger outside the build (I mean in all cases) then I need to run it from the pipeline itself, and when this code is running, the pipeline isn't finished.

Reply:
ah, gotcha, good point! 😄